### PR TITLE
Check that info.branchName exists before working with it

### PR DESCRIPTION
**Pull request by notjosh**
This should generically address https://github.com/greenkeeperio/greenkeeper-lockfile/issues/30 and any similar issues.

If it's not very obviously a greenkeeper branch (because of no env var), then just return early.

### CODE REVIEW

File: update.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ module.exports = function update () {
     return console.error('This build should not update the lockfile. It could be a PR, not a branch build.')
   }
 
+  if (info.branchName == null) {
```
Comment (Realtin) [inline update.js]: What do you think about `(!info.branchName)` ?
so it wouldn't matter if the env variable is `''` or `undefined`

Comment (notjosh) [inline update.js]: My thinking was that `''` would at least work with `.startsWith()` later down the code, and I didn't want to introduce any new behaviour.

But your suggestion looks good. Happy to make the change if you think it makes sense.

Comment (janl) [inline update.js]: +1 I think in this case it is safe to bail early.

Comment (notjosh) [inline update.js]: Updated in aeceb12!

**Pull request merged**